backend/lambdas/custom_resources/redeploy_apigw.py

```diff
@@ -0,0 +1,31 @@
+import boto3
+from crhelper import CfnResource
+from decorators import with_logging
+
+
+helper = CfnResource(json_logging=False, log_level="DEBUG", boto_level="CRITICAL")
+
+api_client = boto3.client("apigateway")
+
+
+@with_logging
+@helper.create
+@helper.delete
+def create(event, context):
+    return None
+
+
+@with_logging
+@helper.update
+def update(event, context):
+    props = event["ResourceProperties"]
+    props_old = event["OldResourceProperties"]
+    if props_old["DeployCognito"] != props["DeployCognito"]:
+        api_client.create_deployment(
```

Comment:
for my own understanding, what's the need to do this ourselves?

Reply:
It turns out that an update to change the auth of an an API does update the api but does not publish that new version so if the deployment is updated to change it this is required otherwise the auth of the API gets out of sync. New deployments work fine it is only the update with change that is affected.